web/modules/custom/joinup_bundle_class/src/JoinupBundleClassFieldAccessTrait.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\joinup_bundle_class;
+
+use Drupal\Core\Field\FieldItemInterface;
+use Drupal\Core\TypedData\Exception\MissingDataException;
+
+/**
+ * Reusable methods for accessing fields in entity bundle classes.
+ */
+trait JoinupBundleClassFieldAccessTrait {
```

Comment:
Creating a module just to provide these wrappers feels excessive. For the event use-case the methods could simply use the entity/field API:

```php
public function getWebUrl(): ?Url {
  $field = $this->get('field_event_web_url')'
  return !$field->isEmpty() ? $field->getUrl() : NULL;
}

public function getLocation(): ?string {
  return $this->get('field_location')->value;
}
```
Note that in both cases you are already acknowledge the type of the fields (see the method's return typing). Knowing the field type you know the property that you want.

At least, I'm sure I'll never use them as I'm so used with the standard API. Probably I'll forgot that they exist after this QA. My recommendation is to not add this wrapper but I will not oppose adding it.

Reply:
But `isEmpty()` is not enough, there might be a `MissingDataException`. Also `EntityInterface::get()` can return NULL, so we need to check for it. So we will end up with the code from the trait anyway, but then duplicated in every field getter.